### ISSUE TITLE
Blue color issues with Sony A7IV files in highlights

### ISSUE THREAD

**Issue by Darkdriver84** [darktable-org/darktable#13200]
I was hoping that the new 4.2. version would address this issues, but it is still present in my files. It seems that highlights have a blue color tone, which cannot even be fixed with the new highlight correction options completely. After tweaking the image with color calibration and white balance the only option I have is to reduce the saturation for blue to get the highlights neutral white.
This behaviour is very strange and annoying at the same time as well as the lack of lossless compression support for Sony. Hope there is a fix very soon.
![_7IV5992](https://user-images.githubusercontent.com/54710750/209533073-294aa4ca-cb21-43f1-8da8-dd2178157c36.jpg)

**Comment by Nilvus** [comment]
A raw and xmp will help. Please add them.

**Comment by Darkdriver84** [comment]
I have uploaded a dng now (since I cannot use the lossless raw) of this photo. You can download it withhin a week from now: https://www.filemail.com/d/eznkxukrhtegugc

**Comment by kmilos** [comment]
And you have already followed the discussions in https://github.com/darktable-org/darktable/issues/11323, https://github.com/darktable-org/rawspeed/pull/349 and https://github.com/darktable-org/rawspeed/pull/385, and tried getting the original ARW file custom black & white levels (using e.g. exiftool) and setting those manually?

**Comment by Darkdriver84** [comment]
> And you have already followed the discussions in #11323, [darktable-org/rawspeed#349](https://github.com/darktable-org/rawspeed/pull/349) and [darktable-org/rawspeed#385](https://github.com/darktable-org/rawspeed/pull/385), and tried getting the original ARW file custom black & white levels (using e.g. exiftool) and setting those manually?

No idea what I should do. It should just work straightforward as it did with my A7 III. I am not into technical details since other software obviously has not this issue.

**Comment by jenshannoschwalm** [comment]
> No idea what I should do.

ATM we still just want a raw file from your camera to analyse if there is something wrong as suggested above :-)

1. A dng file is not relevant at all as we can't and also would not want to check your conversion.
2. You should have a look atv the mentioned reports

Otherwise this issue will be very silent ...

**Comment by Darkdriver84** [comment]
> > No idea what I should do.
> 
> ATM we still just want a raw file from your camera to analyse if there is something wrong as suggested above :-)
> 
>     1. A dng file is not relevant at all as we can't and also would not want to check your conversion.
> 
>     2. You should have a look atv the mentioned reports
> 
> 
> Otherwise this issue will be very silent ...

I will upload an ARW file when I am back from vacation in two days. Have only copied the DNGs for editing on my laptop.

**Comment by Darkdriver84** [comment]
As promised here is the ARW file: https://www.filemail.com/d/zgeisiphloijqjs

What I have noticed is that files opened for the first time with version 4.2 the color calibration is turned off while in version 4.0 color calibration as well as white balance were turned on at the same time.

**Comment by jenshannoschwalm** [comment]
Well, the model `ILCE-7M4` seems to be not supported yet in dt (can't open the raw file ...) at least not in the mode of the raw file you shared.

Also - there are already discussions/pr's about the color matrix as mentioned above by Milos.

Please concider to close this issue as a duplicate, time will solve your problem for sure.

**Comment by kmilos** [comment]
I think the OP is aware that Sony lossless mode is currently not supported, hence the conversion to DNG.

I have now treid my own conversion, and saw something potentialy interesting: while the `BlackLevel` is identical between the DNG and ARW files, I found somwehere very deep in the ARW makernotes
```
WhiteLevel                      : 15360 15360 15360
```
However, adjusting this in the raw black/white point module didn't make a significant difference, I thought it would, so something else is going on, sorry...

P.S. Using the temporary "rename .ARW to .CR3" trick to make it import via libraw, the result is the same (but libraw digs out 15360 from the makernotes), so that eliminates both rawspeed and the DNG conversion.

When you pull down the exposure there is a lot of tail in the histogram blue channel only (real? must be some artificial light source?), so I guess highlight reconstruction struggles with that...

**Comment by Darkdriver84** [comment]
Maybe churches with its windows are not very simple situations, but the blue color issue also appeared with street lamps in long exposure shots at night. Maybe the LED causes problems. I don't know, but I cannot believe that current technology struggles with such issues which are not new to this world and started to appear in version 4.0 since I had to convert my files to DNGs due to compatibility issues. Unfortunately my recent photos I have edited only in version 4.2 have not had any compareable situations. 

Feel free to post my ARW or DNG to any site, which can help to improve the result for others. I am sure I am not the only Sony shooter out there and if I can contribute something to the further development I will be happy.

**Comment by kmilos** [comment]
> Feel free to post my ARW or DNG to any site, which can help to improve the result for others.

I'm afraid only you can do this, as you need to transfer your rights at the same time: https://discuss.pixls.us/t/about-the-play-raw-category/11956

**Comment by todd-prior** [comment]
Can you profile your camera?? Might help... You can cheat and use rec2020 as an input profile... This works not bad with weird blues but its a hack. It looks like making an attempt to translate that you are using the basecurve... this is not a highlight friendly approach usually if you have some overexposed areas.  You could experiment with the color preservation modes and filmic and sigmoid tonemappers... Your first example though likely is challenging with lighting and getting the wb just right... and then I think a lot of that color is reflected/refracted light coming in through the stain glass. When you look at what shows as raw exposure there is not much and it looks like it gets fixed. Blown but could get some gamut help with some other methods. The second one looks like maybe you were using clipped highlights as a HLR method and it can produce these in the highlights... usually the new methods and using the tone eq will handle that??

**Comment by todd-prior** [comment]
> I have uploaded a dng with xmp now (since I cannot use the lossless raw) of the previous photo. You can download it withhin a week from now: https://www.filemail.com/d/eznkxukrhtegugc
> 
> ![qMKyMsyySt](https://user-images.githubusercontent.com/54710750/209552142-2bc88b5b-8db3-4c06-82e3-9f7fe6eea863.jpg) you can see it very clearly in the windows in this screenshot (not edited photo).

This is not the same image as the first one and not the one you uploaded is it??

**Comment by Darkdriver84** [comment]
> > I have uploaded a dng with xmp now (since I cannot use the lossless raw) of the previous photo. You can download it withhin a week from now: https://www.filemail.com/d/eznkxukrhtegugc
> > ![qMKyMsyySt](https://user-images.githubusercontent.com/54710750/209552142-2bc88b5b-8db3-4c06-82e3-9f7fe6eea863.jpg) you can see it very clearly in the windows in this screenshot (not edited photo).
> 
> This is not the same image as the first one and not the one you uploaded is it??

No, this is another image showing what I was talking about. I have many similar photos.

But I have just tried something. I opened the photo mentioned by you in Luminar. Neither the ARW (which is supported) nor the DNG show blue color in the highlights. 
![pBdGKB3lmQ](https://user-images.githubusercontent.com/54710750/210068167-95bac91d-69bb-400e-a4f8-7c1df95a49ba.jpg)

Then I tried to open the image again in darktable and checked where the blue color seems to appear for the first time. In the stack until highlight reconstruction everything is ok, but the color calibration module seems to screw it:
![highlights](https://user-images.githubusercontent.com/54710750/210068322-1474382f-a259-43bd-b598-b34aad12b788.jpg)
![color_calibration](https://user-images.githubusercontent.com/54710750/210068362-da25591d-87cc-4bcd-af15-ae0348d05c79.jpg)

Means that the error is in Darktables color calibration module and it is Darktable and not the image as other software seems to show the image correctly.

**Comment by todd-prior** [comment]
Are you using the basecurve in DT??  Also CC depend on good D65 values coming from the WB module.. What if you simply resort to using legacy WB. If this fixes it then it is those values… The other software is not doing a CAT for wb you may find that simply fixes it…. So you may need to tweak a few things to be comparing apples to apples..

**Comment by Darkdriver84** [comment]
> Are you using the basecurve in DT?? Also CC depend on good D65 values coming from the WB module.. What if you simply resort to using legacy WB. If this fixes it then it is those values… The other software is not doing a CAT for wb you may find that simply fixes it…. So you may need to tweak a few things to be comparing apples to apples.. >

I have not changed any standard setting. I open a file with DT and start editing. Whatever is preset, stays preset. So basecurve has been activated as standard. As I say, it is obviously color calibration not WB module.

**Comment by todd-prior** [comment]
> I have not changed any standard setting. I open a file with DT and start editing. Whatever is preset, stays preset. So basecurve has been activated as standard. As I say, it is obviously color calibration not WB module.

So this can't be true unless they changed it... So defaults would be filmic and modern wb. You have the basecurve by default and its setting the module order to legacy somehow??  THis would not be the defaults for 4.2 I am certain. Its easy to see in preferences if there is a white dot then its been changed... Module order should be default not legacy....

**Comment by todd-prior** [comment]
> As I say, it is obviously color calibration not WB module

CC gets its data from the reference values in the wb module if you don't have it set to camera reference and if those values are not good then CC will not yeild good results... I think Sony has shown to be a bit too red in the D65 reference values.. there is a procedure to shoot an image a a white screen on your monitor to check and tweak these slightly....

**Comment by Darkdriver84** [comment]
> > I have not changed any standard setting. I open a file with DT and start editing. Whatever is preset, stays preset. So basecurve has been activated as standard. As I say, it is obviously color calibration not WB module.
> 
> So this can't be true unless they changed it... So defaults would be filmic and modern wb. You have the basecurve by default and its setting the module order to legacy somehow?? THis would not be the defaults for 4.2 I am certain. Its easy to see in preferences if there is a white dot then its been changed... Module order should be default not legacy....

Thank you for this comment and hint. I changed it in the settings (guess all the updating has never changed this setting) and it seems to solve the issue with the color calibration. Only sacrifice is that I cannot just apply the setting to previously edited photos. But It is alright as I could get rid of the blue color by editing.

**Comment by todd-prior** [comment]
> Only sacrifice is that I cannot just apply the setting to previously edited photos.

Well you can use shift ctrl c and shift ctrl v to copy and paste any part of a history stack to new images so for example you could revert to traditional wb as shot by pasting color calibration as deactivated and wb set to as shot.... set the mode to append and paste just that....assuming this is what you have changed... I would also experiment with the sigmoidal curve and the scene referred workflow... looking at the dng you shared many of the ceiling areas are a bit blown and lacking in details that can easily be recovered ... For many images you can edit then using just exposure, filmic or sigmoid, tone eq and rgb colorbalance to set color grade.... Of course the usually denoise and lens correction stuff and cropping but it is in most cases a superior result to the basecurve that you are using... If you want to stay with that then check out the color preservation modes and they can give a much different look. The default is luminance which can be a bit harsh ..I find checking no or power norm are worth it... Finally consider checking your raw white point... I think this is mentioned above.. Your DNG is way off. If you try the trick mentioned and load your Sony file by adding the CR3 extension you will see it shoudl be around 15363 not 16348 or what ever and this is used by all the HLR methods to manage highlight reconstruction... A value set too high will not allow them to reconstruct properly as white is not correctly defined... Just some random thoughts.... THe raw white point is in the exif data and I am not sure if it changes with iso for your camera or not but you could check.. If not you could create a preset to set it to 15363...

**Comment by todd-prior** [comment]
> Well, with the new settings I should not use the base curve anymore. And yes, the highlights were blown out. Have corrected this already.

- Sounds like progress. As I wasn't there i have no memory for the photo but I did a quick 2 min edit from scratch using filmic, sigmoid. replacing your base curve in your edit with sigmoid and some exposure and finally your edit exported. Using the tone mappers and some tone eq if you look at the ceiling and lights on the wall they get managed a bit better... but wb might be off and I did not attempt to do anything with the colored reflections in from the glass... but just thought might be interesting for demonstration
[_7IV5992_yours+sigmoid.zip](https://github.com/darktable-org/darktable/files/10325857/_7IV5992_yours%2Bsigmoid.zip)
